File: backend/GameApp.Domain/ValueObjects/Scenes/NothingHappensScene.cs
using GameApp.Domain.Enumerates;

namespace GameApp.Domain.ValueObjects.Scenes;

// Class for scene where nothing happens you just move forward
public class NothingHappensScene : Scene
{

    // Constructor
    public NothingHappensScene(SceneName name, SceneDescription description, Biomes biome)
    : base(name, description, biome)
    {

    }

    // Setters 
    public NothingHappensScene SetSceneName(SceneName newName) =>
    new NothingHappensScene(newName, GetDescription(), GetBiome());

    public NothingHappensScene SetSceneDescription(SceneDescription newDescription) =>
    new NothingHappensScene(GetName(), newDescription, GetBiome());

    public NothingHappensScene SetBiome(Biomes newBiome) =>
    new NothingHappensScene(GetName(), GetDescription(), newBiome);

    // To string
    public override string ToString()
    {
        return $"{GetName()} NothingHappens Scene: " +
               $"Description={GetDescription()}, Biome={GetBiome()}, ";
    }

}

File: frontend/gameapp-frontend/src/domain/entities/scenes/nothing-happens-scene.js
import { Scene } from "./scene.js";

// NothingHappensScene
export class NothingHappensScene extends Scene {
    constructor(name, description, biome, id = null) {
        super(name, description, biome, id);
    }

    // setter
    setSceneName(newName) { return new NothingHappensScene(newName, this._description, this._biome, this._id); }
    setSceneDescription(newDescription) { return new NothingHappensScene(this._name, newDescription, this._biome, this._id); }
    setBiome(newBiome) { return new NothingHappensScene(this._name, this._description, newBiome, this._id); }

    toString() {
        return `${this._name} NothingHappens Scene(${this._id}): Description=${this._description}, Biome=${this._biome}`;
    }
}